dataloaden.go

```diff
@@ -1,28 +1,217 @@
-package main
+package dataloaden
 
 import (
-	"fmt"
-	"os"
-
-	"github.com/vektah/dataloaden/pkg/generator"
+	"sync"
+	"time"
 )
 
-func main() {
-	if len(os.Args) != 4 {
-		fmt.Println("usage: name keyType valueType")
-		fmt.Println(" example:")
-		fmt.Println(" dataloaden 'UserLoader int []*github.com/my/package.User'")
-		os.Exit(1)
+// LoaderConfig captures the config to create a new Loader
+type LoaderConfig[K comparable, T any] struct {
+	// Fetch is a method that provides the data for the loader
+	Fetch func(keys []K) ([]T, []error)
+
+	// Wait is how long wait before sending a batch
```

Comment:
Is there a typo here? i.e. 
```suggestion
	// Wait is how long to wait before sending a batch
```